test/command/test-check-performance-regression.rb

```diff
@@ -0,0 +1,272 @@
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+require "groonga-query-log/command/check-performance-regression"
+
+class CheckPerformanceRegressionCommandTest < Test::Unit::TestCase
+  include Helper::Path
+
+  def setup
+    @command = GroongaQueryLog::Command::CheckPerformanceRegression.new
+  end
+
+  def fixture_path(*components)
+    super("check-performance-regression", *components)
+  end
+
+  sub_test_case("options") do
+
+    MISSING_QUERY_LOG_ERROR = <<-OUTPUT
+query logs is not specified. use --input-old-query and --input-new-query
+    OUTPUT
+
+    def run_command_with_stderr
+      actual = ""
+      Tempfile.open("redirect-stderr-tmpfile") do |file|
+        $stderr.reopen(file)
+        yield(file)
+        $stderr.flush
+        file.rewind
+        actual = file.read
+      end
+      actual
+    end
+
+    def test_no_option
+      actual = run_command_with_stderr do
+        @command.run([])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_no_input_old_query
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-new-query=" + fixture_path("query2.log")
+        ])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_no_input_new_query
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-old-query=" + fixture_path("query1.log")
+        ])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_nonexistent_input_old_query
+      expected = <<-OUTPUT
+invalid option: --input-old-query=#{fixture_path("nonexsistent.log")}
+      OUTPUT
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-old-query=" + fixture_path("nonexsistent.log")
+        ])
+      end
+      assert_equal(expected, actual)
+    end
+
+    def test_nonexistent_input_new_query
+      expected = <<-OUTPUT
+invalid option: --input-new-query=#{fixture_path("nonexsistent.log")}
+      OUTPUT
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-new-query=" + fixture_path("nonexsistent.log")
+        ])
+      end
+      assert_equal(expected, actual)
+    end
+
+    def test_output
+      path = "/tmp/output"
```

Comment:
We should use `Tempfile` for this propose.

Reply:
Fixed to use Tempfile.

Reply:
I've fixed not to conflict `open_output(..)` block.